inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -159,38 +235,79 @@ std::vector<PointSpeedPair> InLaneCruisingPlugin::constrain_to_time_boundary(con
   return time_bound_points;
 }
 
+int InLaneCruisingPlugin::get_nearest_point_index(const std::vector<lanelet::BasicPoint2d>& points,
+                                               const cav_msgs::VehicleState& state)
+{
+  lanelet::BasicPoint2d veh_point(state.X_pos_global, state.Y_pos_global);
+  double min_distance = std::numeric_limits<double>::max();
+  int i = 0;
+  int best_index = 0;
+  for (const auto& p : points)
+  {
+    double distance = lanelet::geometry::distance2d(p, veh_point);
+    if (distance < min_distance)
+    {
+      best_index = i;
+      min_distance = distance;
+    }
+    i++;
+  }
+
+  return best_index;
+}
+
+
 std::vector<cav_msgs::TrajectoryPlanPoint> InLaneCruisingPlugin::compose_trajectory_from_centerline(
     const std::vector<PointSpeedPair>& points, const cav_msgs::VehicleState& state)
 {
-  ROS_DEBUG_STREAM("VehicleState: "
+ ROS_DEBUG_STREAM("VehicleState: "
                    << " x: " << state.X_pos_global << " y: " << state.Y_pos_global << " yaw: " << state.orientation
                    << " speed: " << state.longitudinal_vel);
 
   ROS_DEBUG_STREAM("points size: " << points.size());
   
-  log::printDebugPerLine(points, &log::pointSpeedPairToStream);
+  //log::printDebugPerLine(points, &log::pointSpeedPairToStream);
```

Comment:
Lets either delete it or, uncomment it if this print still has value for debugging.

Reply:
I am thinking of keeping it